### PR TITLE
docs: Updated docs for v12 release

### PR DESCRIPTION
**Pull request by mmulji-ic**
## Description

Part of epic #2699 

Updated docs for v12 release.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct `docs:` prefix in the PR title
- [x] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] provided a link to the relevant issue or specification
- [x] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] Confirmed the correct `docs:` prefix in the PR title
- [ ] Confirmed all author checklist items have been addressed 
- [ ] Confirmed that this PR only changes documentation
- [ ] Reviewed content for consistency
- [ ] Reviewed content for thoroughness
- [ ] Reviewed content for spelling and grammar
- [ ] Tested instructions (if applicable)

### CODE REVIEW

**Review by mpoke — APPROVED** [review]

File: docs/migration/cosmoshub-4-v12-upgrade.md [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+### Target runtime
+
+The Cosmos Hub mainnet network, `cosmoshub-4`, will run [Gaia v12.0.0](https://github.com/cosmos/gaia/releases/tag/v12.0.0). Operators _**MUST**_ use this version post-upgrade to remain connected to the network.
+
+## Upgrade steps
+
+There are 2 major ways to upgrade a node:
+
+- Manual upgrade
+- Upgrade using [Cosmovisor](https://github.com/cosmos/cosmos-sdk/tree/master/cosmovisor)
```
Comment (dasanchez) [inline docs/migration/cosmoshub-4-v12-upgrade.md]:
```suggestion
- Upgrade using [Cosmovisor](https://pkg.go.dev/cosmossdk.io/tools/cosmovisor)
```
The github.com link is not working anymore.

File: docs/migration/cosmoshub-4-v12-upgrade.md [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+### Method II: Upgrade using Cosmovisor
+
+### Manually preparing the binary
+
+##### Preparation
+
+Install the latest version of Cosmovisor (`1.5.0`):
+
+```shell
+go install github.com/cosmos/cosmos-sdk/cosmovisor/cmd/cosmovisor@latest
```
Comment (dasanchez) [inline docs/migration/cosmoshub-4-v12-upgrade.md]:
```suggestion
go install cosmossdk.io/tools/cosmovisor/cmd/cosmovisor@latest
```
They changed the package with v1.4.0

File: docs/migration/cosmoshub-4-v12-upgrade.md [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+### Auto-Downloading the Gaia binary
+
+**This method is not recommended!**
+
+#### Preparation
+
+Install the latest version of Cosmovisor (`1.5.0`):
+
+```shell
+go install github.com/cosmos/cosmos-sdk/cosmovisor/cmd/cosmovisor@latest
```
Comment (dasanchez) [inline docs/migration/cosmoshub-4-v12-upgrade.md]:
```suggestion
go install cosmossdk.io/tools/cosmovisor/cmd/cosmovisor@latest
```

**Review by dasanchez — COMMENTED** [review]
I updated a couple of Cosmovisor links, otherwise looks good 👍🏽

**Review by MSalopek — APPROVED** [review]

**Pull request merged**